src/Carting.Api/Controllers/V1/CartingController.cs

```diff
@@ -0,0 +1,138 @@
+    /// </summary>
+    /// <param name="cartId">Identifier of the cart.</param>
+    /// <returns>Cart items for the provided cartId.</returns>
+    /// <remarks>
+    /// Sample request:
+    ///
+    ///     GET /cart/1
+    ///
+    /// </remarks>
+    /// <response code="200">Cart items returned successfully</response>
+    /// <response code="500">Internal server error occured</response>
+    [HttpGet]
+    [Route("cart/{cartId}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public IActionResult Get(string cartId)
+    {
+        try
+        {
+            var result = cartingService.GetCartItems(cartId);
+
+            return Ok(Mapper.Map(cartId, result));
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, ex.Message);
+        }
+    }
+
+    /// <summary>
+    /// Creates a cart item in the cart for the provided cartId.
+    /// </summary>
+    /// <param name="cartId">Identifier of the cart.</param>
+    /// <param name="request">Cart item request.</param>
+    /// <returns></returns>
+    /// <remarks>
+    /// Sample request:
+    ///
+    ///     POST /Cart/1
+    ///     {
+    ///        "name": "string",
+    ///        "image": {
+    ///             "url": "string",
+    ///             "alt": "string",
+    ///        },
+    ///        "price": 0,
+    ///        "quantity": 0
+    ///     }
+    ///
+    /// </remarks>
+    /// <response code="200">Cart items returned successfully</response>
+    /// <response code="400">Bad request</response>
+    /// <response code="500">Internal server error occured</response>
+    [HttpPost]
+    [Route("cart/{cartId}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public IActionResult Post(string cartId, CartItemRequest request)
+    {
+        try
+        {
+            cartingService.AddCartItem(Mapper.Map(cartId, request));
+
+            return Ok();
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, ex.Message);
+        }
+    }
+
+    /// <summary>
+    /// Deletes a cart item from the cart based on the provided cartId.
+    /// </summary>
+    /// <param name="cartId">Identifier of the cart.</param>
+    /// <param name="itemId">Identifier of the cart item.</param>
+    /// <returns></returns>
+    /// <remarks>
+    /// Sample request:
+    ///
+    ///     DELETE /cart/1?itemId=1
+    ///
+    /// </remarks>
+    /// <response code="200">Cart item deleted successfully</response>
+    /// <response code="404">Cart or cart item not found</response>
+    /// <response code="500">Internal server error occured</response>
+    [HttpDelete]
+    [Route("cart/{cartId}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
```

Comment:
Could you point me to a code returning 404

Reply:
https://github.com/bertalan-kiss/carting/blob/feature/module4-task2/src/Carting.Api/Controllers/V1/CartingController.cs#L130